cli/stream_command.go

```diff
@@ -1403,18 +1415,61 @@ func (c *streamCmd) copyAndEditStream(cfg api.StreamConfig, pc *fisk.ParseContex
 		return cfg, fmt.Errorf("invalid compression algorithm")
 	}
 
-	if c.repubDest != "" && c.repubSource != "" {
-		cfg.RePublish = &api.RePublish{
-			Source:      c.repubSource,
-			Destination: c.repubDest,
-			HeadersOnly: c.repubHeadersOnly,
-		}
+	var repubConfig api.RePublish
+
+	if cfg.RePublish != nil {
+		repubConfig = *cfg.RePublish
+	} else {
+		repubConfig = api.RePublish{}
+	}
+
+	if c.repubSource != "" {
+		repubConfig.Source = c.repubSource
+	}
+	if c.repubDest != "" {
+		repubConfig.Destination = c.repubDest
+	}
+	if c.repubHeadersOnly {
+		repubConfig.HeadersOnly = true
+	} else if repubConfig.HeadersOnly {
+		repubConfig.HeadersOnly = false
+	}
+
+	if (repubConfig.Source == "" || repubConfig.Source == ">") && repubConfig.Destination == "" {
+		cfg.RePublish = nil
+	} else {
+		cfg.RePublish = &repubConfig
 	}
 
 	if c.noRepub {
 		cfg.RePublish = nil
 	}
 
+	var subjectTransformConfig api.SubjectTransformConfig
+
+	if cfg.SubjectTransform != nil {
+		subjectTransformConfig = *cfg.SubjectTransform
+	} else {
+		subjectTransformConfig = api.SubjectTransformConfig{}
+	}
+
+	if c.subjectTransformSource != "" {
+		subjectTransformConfig.Source = c.subjectTransformSource
+	}
+	if c.subjectTransformDest != "" {
+		subjectTransformConfig.Destination = c.subjectTransformDest
+	}
+
+	if subjectTransformConfig.Source == "" && subjectTransformConfig.Destination == "" {
+		cfg.SubjectTransform = nil
+	} else {
+		cfg.SubjectTransform = &subjectTransformConfig
+	}
+
+	if c.noSubjectTransform {
```

Comment:
You are right that some of it is not needed anymore after I added noSubjectTransform, which is so that you can remove a mapping using stream edit. Originally it tried to assume stream edit w/out specifying a transform source or dest would mean 'remove the transform', but you can't use the absence of the argument to assume you want to not have any filter (incompatible with multiple filters/transforms anyways) hence now there's noSubjectTransform added to edit. Unnecessary code removed.